src/style_builder.cr

```diff
@@ -192,14 +214,14 @@ module Mint
       false
     end
 
-    def prefixed_class_name(node)
-      @css_prefix.to_s + style_pool.of(node, nil)
+    def prefixed_class_name(node : Ast::Style, id : String? = nil)
+      @css_prefix.to_s + style_pool.of(node, id)
     end
 
     # The main entry point for processing a "style" tag.
-    def process(node : Ast::Style)
+    def process(node : Ast::Style, id : String)
```

Comment:
Shouldn't this allow for nil-able `id` too?

```suggestion
    def process(node : Ast::Style, id : String? = nil)
```

Reply:
It compiles so _I guess not_?

Reply:
I'm asking since the API all around it allows for empty `id`, so I don't see a reason why this one shouldn't.

Reply:
It calls `prefixed_class_name` which has nil-able `id`, this, in turn, invokes `StylePool#of` which does the same, etc.

Reply:
> It compiles so _I guess not_?

Just because there are no tests stressing this path :P